src/System.Net.Sockets/src/System/Net/Sockets/TCPClient.Windows.netstandard17.cs

```diff
@@ -0,0 +1,135 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Threading.Tasks;
+
+namespace System.Net.Sockets
+{
+    partial class TcpClient
+    {
+        private void ConnectCore(string hostname, int port)
+        {
+            // IPv6: We need to process each of the addresses return from
+            //       DNS when trying to connect. Use of AddressList[0] is
+            //       bad form.
+            IPAddress[] addresses = Dns.GetHostAddresses(hostname);
+            Exception lastex = null;
+            Socket ipv6Socket = null;
+            Socket ipv4Socket = null;
+
+            try
+            {
+                if (_clientSocket == null)
+                {
+                    if (Socket.OSSupportsIPv4)
+                    {
+                        ipv4Socket = new Socket(AddressFamily.InterNetwork, SocketType.Stream, ProtocolType.Tcp);
+                    }
+                    if (Socket.OSSupportsIPv6)
+                    {
+                        ipv6Socket = new Socket(AddressFamily.InterNetworkV6, SocketType.Stream, ProtocolType.Tcp);
```

Comment:
Yes, though I'd like to address that separately, when I have better tests to validate the change. :)